lib/src/error_tracking/dart_exception_processor.dart

```diff
@@ -0,0 +1,335 @@
+      isGeneratedStackTrace = true; // Flag to remove top PostHog frames
+    }
+
+    // Process single exception (Dart doesn't provide standard exception chaining afaik)
+    final frames = _parseStackTrace(
+      effectiveStackTrace,
+      inAppIncludes: inAppIncludes,
+      inAppExcludes: inAppExcludes,
+      inAppByDefault: inAppByDefault,
+      removeTopPostHogFrames: isGeneratedStackTrace,
+    );
+
+    // we consider primitives and generated Strack traces as synthetic
+    final exceptionData = <String, dynamic>{
+      'type': _getExceptionType(error),
+      'mechanism': {
+        'handled': handled,
+        'synthetic': _isPrimitive(error) || isGeneratedStackTrace,
+        'type': 'generic',
+      },
+      'thread_id': _getCurrentThreadId(),
+    };
+
+    // Add exception message, if available
+    final errorMessage = error.toString();
+    if (errorMessage.isNotEmpty) {
+      exceptionData['value'] = errorMessage;
+    }
+
+    // Add module from first stack frame (where exception was thrown)
+    final exceptionModule = _getExceptionModule(effectiveStackTrace);
+    if (exceptionModule != null && exceptionModule.isNotEmpty) {
+      exceptionData['module'] = exceptionModule;
+    }
+
+    // Add package from first stack frame (where exception was thrown)
+    final exceptionPackage = _getExceptionPackage(effectiveStackTrace);
+    if (exceptionPackage != null && exceptionPackage.isNotEmpty) {
+      exceptionData['package'] = exceptionPackage;
+    }
+
+    // Add stacktrace, if any frames are available
+    if (frames.isNotEmpty) {
+      exceptionData['stacktrace'] = {
+        'frames': frames,
+        'type': 'raw',
+      };
+    }
+
+    final result = <String, dynamic>{
+      '\$exception_level': handled ? 'error' : 'fatal',
+      '\$exception_list': [exceptionData],
+    };
+
+    // Add custom properties if provided
+    if (properties != null) {
+      for (final entry in properties.entries) {
+        // Don't allow overwriting system properties
+        if (!result.containsKey(entry.key)) {
+          result[entry.key] = entry.value;
+        }
+      }
+    }
+
+    return result;
+  }
+
+  /// Determines if a stack frame belongs to PostHog SDK (just check package for now)
+  static bool _isPostHogFrame(Frame frame) {
+    return frame.package == 'posthog_flutter';
+  }
+
+  /// Parses stack trace into PostHog format
+  static List<Map<String, dynamic>> _parseStackTrace(
+    StackTrace stackTrace, {
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+    bool removeTopPostHogFrames = false,
+  }) {
+    final chain = Chain.forTrace(stackTrace);
+    final frames = <Map<String, dynamic>>[];
+
+    for (final trace in chain.traces) {
+      bool skipNextPostHogFrame = removeTopPostHogFrames;
+
+      for (final frame in trace.frames) {
+        // Skip top PostHog frames?
+        if (skipNextPostHogFrame) {
+          if (_isPostHogFrame(frame)) {
```

Comment:
we should only skip if we generated the stack trace, otherwise it should be there because it was likely a bug on our end or something